crates/sui-core/src/consensus_adapter.rs

```diff
@@ -234,18 +251,23 @@ impl ConsensusAdapter {
     /// Authorities higher in the list wait less time.
     ///
     /// The function targets having only 1 consensus transaction submitted per user transaction
-    /// when system operates normally
+    /// when system operates normally.
+    ///
+    /// The function returns the delay to wait and the position of authority in the list.
     fn submit_delay_certificate(
         committee: &Committee,
         ourselves: &AuthorityName,
         tx_digest: &TransactionDigest,
-    ) -> Duration {
+    ) -> (Duration, usize) {
         let position = position_submit_certificate(committee, ourselves, tx_digest);
         const MAX_DELAY_MUL: usize = 10;
         // DELAY_STEP is chosen as 1.5 * mean consensus delay
         // In the future we can actually use information about consensus rounds instead of this delay
         const DELAY_STEP: Duration = Duration::from_secs(7);
-        DELAY_STEP * std::cmp::min(position, MAX_DELAY_MUL) as u32
+        (
+            DELAY_STEP * std::cmp::min(position, MAX_DELAY_MUL) as u32,
```

Comment:
Should we use `min(position, MAX_DELAY_MUL)` as the position for metrics instead? I think we are interested more in delay than actual positions. This would cap the cardinality too.

Reply:
I think the position is useful as well, since we may have variable delay times soon